point_viewer_grpc/src/service.rs

```diff
@@ -245,19 +255,53 @@ impl proto_grpc::Octree for OctreeService {
     fn get_all_points(
         &self,
         ctx: RpcContext,
-        _req: proto::GetAllPointsRequest,
+        req: proto::GetAllPointsRequest,
         resp: ServerStreamingSink<proto::PointsReply>,
     ) {
-        stream_points_back_to_sink(OctreeQuery::FullPointcloud, self.octree.clone(), &ctx, resp)
+        stream_points_back_to_sink(
+            OctreeQuery::FullPointcloud,
+            Arc::clone(&self.get_service_data(&req.octree_id)),
+            &ctx,
+            resp,
+        )
     }
 }
 
-pub fn start_grpc_server(octree_directory: &Path, host: &str, port: u16) -> Server {
+impl OctreeService {
+    pub fn get_service_data(&self, octree_id: &str) -> Arc<OctreeServiceData> {
+        match self.data_cache.read().unwrap().get(octree_id) {
+            None => {
+                let octree = self
+                    .factory
+                    .generate_octree(self.location.join(&octree_id).to_string_lossy())
+                    .unwrap();
```

Comment:
I'm a little averse to `unwraps` ;) Do you think we could at least replace these unwraps by `expect` and provide some meaningful message?

Reply:
I implemented proper error handling... see new code.